src/transport/SecureSession.h

```diff
@@ -114,6 +114,18 @@ class SecureSession : public Session
     uint16_t GetPeerSessionId() const { return mPeerSessionId; }
     FabricIndex GetFabricIndex() const { return mFabric; }
 
+    // Should only be called for PASE sessions, which start with undefined fabric,
+    // to place on a newly commissioned fabric after successful
+    // OperationalCredentialsCluster::AddNOC
+    CHIP_ERROR NewFabric(FabricIndex fabricIndex)
+    {
+        // TODO: should check that secure session type is PASE and current value is undefined
+        // (i.e. that it's called exactly once in proper circumstances)
+        // but that's difficult until issue #13711 is addressed
+        mPaseFabric = fabricIndex;
```

Comment:
OK, but `GetFabricIndex` will not start being right as a result, will it?  This will fix things for ACLs, but other consumers of fabric index on the session will continue to be broken.

It seems to me that what we should actually be doing is storing the given index in `mFabric` (which therefore needs to stop being const), but only doing that if we in fact have a PASE session....

Reply:
Yeah I was trying to avoid changing it from const but it looks like I'll have to.

Reply:
It's also possible for GetAccessingFabric to return either of the member variables, appropriately. We'd have discovered this in a few days when trying to actually run stuff. But I'll fix it now, see which way works best.

Reply:
Yes, the high-level bit is that GetFabricIndex() needs to return the right thing.  Note that it's likely this will move up to Session as #13397 is fixed...

Reply:
I changed it to use mFabric which will be OK for now (sans check for PASE) if we aren't actually commissioning over CASE. This gives us a few days to fix #13711. If anyone is actually commissioning over CASE, then I should change it back to use my original technique, special case GetAccessingFabric, and then change it back this way once #13711 is fixed.

Reply:
See my comment here: https://github.com/project-chip/connectedhomeip/pull/13712#discussion_r788183100

We actually worked really hard in the spec to preserve the permissibility of AddNOC occurring over CASE.  The fail-safe context logic was actually somewhat simplified by allowing this.

The key is that the fail-safe context fabric association and accessing fabric do not need to be the same.

Reply:
Yes, I did find the spec text talking about CASE as well as PASE. My question is more specific to this week: is anyone doing AddNOC over CASE now?

Reply:
Ah, OK.  I understand now.  So until #13711 is resolved, there will be a temporary change in functionality with this PR for any out-of-tree code that would attempt AddNOC over CASE (no in-tree code currently does this).

We would not be impacted by this currently.  From my end, resolution as part of of #13711 is sufficient, so my approval stands.

I suppose others can chime in if they object to the PR.

Reply:
Exactly: @emargolis is looking at #13711 to see if it can be easily resolved, so in the mean time this PR lacking the check would INCORRECTLY migrate a CASE session as well as a PASE session, which is OK for just immediate short term if nobody is actually doing that. When that issue is fixed, I add the checks, and if it's deemed to be not easily fixable, I can work around it (which is more work to both work around it, and undo it all later). Hoping that's OK till beginning of next week.

Reply:
I _think_ that's fine, yes. Thank you for clarifying that!